File: server/models/momentComment.js
const { ulid } = require("ulid");

function createMomentComment({ momentId, userId, content, replyTo = null }) {
  const now = new Date().toISOString();

  return {
    momentId,
    commentId: ulid(),
    userId: String(userId),
    content,
    replyTo,
    reactions: [],
    createdAt: now,
    updatedAt: now
  };
}

module.exports = { createMomentComment };

File: docs/visual-paradigm-reverse/src/main/java/com/zalofaker/model/MomentComment.java
package com.zalofaker.model;

import java.util.List;

public class MomentComment {
    public String momentId;
    public String commentId;
    public String userId;
    public String content;
    public String replyToCommentId;
    public String createdAt;
    public String updatedAt;

    public Moment moment;
    public User user;
    public MomentComment replyTo;
    public List<MomentComment> replies;
}
